jena-text/src/test/java/org/apache/jena/query/text/filter/TestSelectiveFoldingFilter.java

```diff
@@ -0,0 +1,135 @@
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        List<String> expected = Arrays.asList("Senora", "Siobhan", "look", "at", "that", "facade");
+        assertTrue(tokens.equals(expected));
+    }
+
+    @Test
+    public void testSingleCharacterWhiteListed() throws IOException {
+        whitelisted = new CharArraySet(Arrays.asList("ç"), false);
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        List<String> expected = Arrays.asList("Senora", "Siobhan", "look", "at", "that", "façade");
+        assertTrue(tokens.equals(expected));
+    }
+
+    @Test
+    public void testCompleteWhiteListed() throws IOException {
+        whitelisted = new CharArraySet(Arrays.asList("ñ", "á", "ç"), false);
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        // here we should have the complete input
+        List<String> expected = Arrays.asList("Señora", "Siobhán", "look", "at", "that", "façade");
+        assertTrue(tokens.equals(expected));
+    }
+
+    @Test
+    public void testCaseMatters() throws IOException {
+        // note the first capital letter
+        whitelisted = new CharArraySet(Arrays.asList("Ñ", "á", "ç"), false);
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        List<String> expected = Arrays.asList("Senora", "Siobhán", "look", "at", "that", "façade");
+        assertTrue(tokens.equals(expected));
+    }
+
+    @Test
+    public void testMismatchWhiteList() throws IOException {
+        whitelisted = new CharArraySet(Arrays.asList("ú", "ć", "ž"), false);
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        List<String> expected = Arrays.asList("Senora", "Siobhan", "look", "at", "that", "facade");
+        assertTrue(tokens.equals(expected));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testNullWhiteListThrowsError() throws IOException {
+        collectTokens(inputText, null);
+    }
+
+    @Test
+    public void testEmptyInput() throws IOException {
+        whitelisted = new CharArraySet(Arrays.asList("ç"), false);
+        inputText = new StringReader("");
+        List<String> tokens = collectTokens(inputText, whitelisted);
+        List<String> expected = Collections.emptyList();
+        assertTrue(tokens.equals(expected));
+    }
+
+    /**
+     * Return the list of CharTermAttribute converted to a list of String's.
+     *
+     * @param whitelisted white-list
+     * @return list of CharTermAttribute converted to a list of String's
+     * @throws IOException from Lucene API
+     */
+    private List<String> collectTokens(StringReader inputText, CharArraySet whitelisted) throws IOException {
+        StandardTokenizer tokenizer = new StandardTokenizer();
+        tokenizer.setReader(inputText);
+
+        // Replace this by a C, and then the ASCIIFolding function will take care to replace it by c, as it was not whitelisted
```

Comment:
Leftover dev comment?

Reply:
You got me @rvesse . Spent some 5 minutes looking at the code... at the comment... at the code... at the comment... and couldn't remember every writing it to be honest :-) definitely some leftover. Nice catch, thanks for reviewing it.